src/linux/namespaces.rs

```diff
@@ -67,8 +68,25 @@ fn create_mount_namespace(bind_mounts: HashMap<PathBuf, MountAttrFlags>) -> Resu
     // aren't created outside the sandbox.
     mount_tmpfs(&new_root_c)?;
 
+    // Canonicalize paths, to resolve symlinks.
+    //
+    // If the working directory cannot be accessed, we ignore relative paths.
+    let mut symlinks = Vec::new();
+    let mut bind_mounts = bind_mounts
+        .into_iter()
+        .filter_map(|(path, exception)| {
+            let canonicalized = path.canonicalize().ok()?;
+
+            // Store original symlink path to create it if necessary.
+            if let Ok(target) = path.read_link() {
+                symlinks.push((path, target));
```

Comment:
There is a bit of a problem here.... Because `target` may not be fully resolved. So it may not be in the sandbox...

```sh-session
# ls -ld /tmp/a /tmp/a/foo.txt /tmp/b /tmp/foo.txt
drwxr-xr-x. 2 root root 21 Oct 26 18:41 /tmp/a
-rw-r--r--. 1 root root  0 Oct 26 18:41 /tmp/a/foo.txt
lrwxrwxrwx. 1 root root  6 Oct 26 18:41 /tmp/b -> /tmp/a
lrwxrwxrwx. 1 root root 14 Oct 26 18:42 /tmp/foo.txt -> /tmp/b/foo.txt
# cargo run --example sandbox -- -e /bin -e /lib -r /tmp/foo.txt /bin/bash
    Finished dev [unoptimized + debuginfo] target(s) in 0.59s
     Running `target/debug/examples/sandbox -e /bin -e /lib -r /tmp/foo.txt /bin/bash`
bash-5.2# ls -l /tmp/
total 0
drwxr-xr-x. 2 0 0 60 Oct 26 18:42 a
lrwxrwxrwx. 1 0 0 14 Oct 26 18:42 foo.txt -> /tmp/b/foo.txt
```
One possible solution would be to use the fully resolved path.

```suggestion
                symlinks.push((path, canonicalized));
```